packetbeat/protos/protos.go

```diff
@@ -173,16 +173,9 @@ func (s ProtocolsStruct) configureProtocol(test bool, device string, pub reporte
 
 	if device != "" {
 		// This could happen earlier, but let any errors be found first.
-		var protocol struct {
-			Device string `config:"interface"`
-		}
-		err := config.Unpack(&protocol)
-		if err != nil {
+		if isValid, err := validateProtocolDevice(device, config); !isValid {
```

Comment:
although I can see that the `validateProtocolDevice` always returns `false` on err, I would still add the `err != nil` in the condition
```suggestion
		if isValid, err := validateProtocolDevice(device, config); !isValid || err != nil {
```

Reply:
In this case, always that `validateProtocolDevice` is `false` we want to return, but not always there will be an error, so by adding that clause the logic would change from the original.

Reply:
I still prefer explicitness; `validateProtocolDevice` returns an err that you don't check and you only rely back only in the case the returned bool is false and this is not explicit. Personally if I do a change in the future in `validateProtocolDevice` and return an error I am expecting the caller to not "trust" the other values as I returned an error

Reply:
Also I don't follow how an `|| err != nil` changes the logic, you still always return when `isValid == false` but also when there is an error